### PR TITLE
Remove unnecessary WaitSetErrorResponse

### PR DESCRIPTION
**Pull request by nnmm**
By doing the upgrade outside of the wait set, this struct can be entirely omitted.

### CODE REVIEW

File: rclrs/src/node/mod.rs [hunk 1]
```diff
@@ -110,4 +110,14 @@ impl Node {
             .push(Arc::downgrade(&subscription) as Weak<dyn SubscriptionBase>);
         Ok(subscription)
     }
+
+    /// Returns the subscriptions that have not been dropped yet.
+    pub(crate) fn live_subscriptions(&self) -> Vec<Arc<dyn SubscriptionBase>> {
+        let live_subs = self
```
Comment (esteve) [inline rclrs/src/node/mod.rs:116]: This can be made more succinct by dropping the `live_subs` variable

File: rclrs/src/lib.rs [hunk 2]
```diff
@@ -81,45 +81,37 @@ pub fn spin(node: &node::Node) -> Result<(), WaitSetErrorResponse> {
...
-            Err(err) => return Err(err),
-        };
+    let live_subscriptions = node.live_subscriptions();
+    for sub in &live_subscriptions {
+        wait_set.add_subscription(&**sub)?;
     }
 
     wait_set.wait(timeout_ns)?;
-    for (i, sub) in node.subscriptions.iter().enumerate() {
+    for (i, sub) in live_subscriptions.iter().enumerate() {
```
Comment (esteve) [inline rclrs/src/lib.rs:108]: Please avoid shortened names, rename `sub` to `live_subscription`

**Review by esteve — CHANGES_REQUESTED** [review]

File: rclrs/src/lib.rs [hunk 3]
```diff
@@ -81,45 +81,37 @@ pub fn spin(node: &node::Node) -> Result<(), WaitSetErrorResponse> {
...
+        &mut *node.context.lock(),
     )?;
 
-    for subscription in &node.subscriptions {
-        match wait_set.add_subscription(subscription) {
-            Ok(()) => (),
-            Err(WaitSetErrorResponse::DroppedSubscription) => (),
-            Err(err) => return Err(err),
-        };
+    let live_subscriptions = node.live_subscriptions();
```
Comment (esteve) [inline rclrs/src/lib.rs:102]: Should this be guarded by a lock in case the stored subscriptions change?

Comment (nnmm) [reply rclrs/src/lib.rs:102]: No, it's a vector of `Arc`s, so as long as this variable is in scope they won't change.

File: rclrs/src/lib.rs (hunk 2 above)
Comment (nnmm) [reply rclrs/src/lib.rs]: Done (though I personally don't mind the shorter names as it is visually more distinct from the `live_subscriptions` variable).

File: rclrs/src/node/mod.rs (hunk 1 above)
Comment (nnmm) [reply rclrs/src/node/mod.rs]: Yep, I at first considered also pruning dead subscriptions in here, but then decided against it. Done.

**Review by esteve — APPROVED** [review]

**Pull request merged**